### PR TITLE
OCPCLOUD-1852: Bump k8s dependencies to 1.26; go 1.19

### PR DESCRIPTION
**Pull request by RadekManak**

### CODE REVIEW

**Comment by RadekManak** [comment]
Just noticed that #1100 also bumps to 1.26 & go 1.19.

**Comment by RadekManak** [comment]
Looking at the e2e-libvirt failure, I think that origin suite needs to update first.

**Review by elmiko — COMMENTED** [review]
generally looks good to me
/lgtm

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -123,11 +123,11 @@ func New(
 		operandVersions: operandVersions,
 	}
 
-	deployInformer.Informer().AddEventHandler(optr.eventHandlerDeployments())
-	daemonsetInformer.Informer().AddEventHandler(optr.eventHandler())
-	validatingWebhookInformer.Informer().AddEventHandler(optr.eventHandlerSingleton(isMachineWebhook))
-	mutatingWebhookInformer.Informer().AddEventHandler(optr.eventHandlerSingleton(isMachineWebhook))
-	featureGateInformer.Informer().AddEventHandler(optr.eventHandler())
+	_, _ = deployInformer.Informer().AddEventHandler(optr.eventHandlerDeployments())
```
Comment (JoelSpeed) [inline pkg/operator/operator.go:126]: What do these even return?

Comment (RadekManak) [reply pkg/operator/operator.go:126]: The first object that can be used to cancel the event handler and error.

**Comment by RadekManak** [comment]
The linter was only setup to run on ./pkg. I have set it up to also run on ./cmd and fixed issues that it found.

File: pkg/operator/operator.go (hunk 1 above)
Comment (RadekManak) [reply pkg/operator/operator.go]: We probably should. Added.

File: cmd/machine-api-operator/start.go [hunk 2]
```diff
@@ -57,7 +57,9 @@ func init() {
 }
 
 func runStartCmd(cmd *cobra.Command, args []string) {
```
Comment (JoelSpeed) [inline cmd/machine-api-operator/start.go:59]: If you change the signature here to return an error, and change `Run: runStartCmd,` to `RunE: runStartCmd` does that make the error handling easier? Means you don't have to use klog.Fatal and panics

**Review by JoelSpeed — COMMENTED** [review]
I think we can improve the structure in cmd, see my comment, otherwise LGTM

**Comment by RadekManak** [comment]
Looks like someone got the k8s & go 1.19 bump into main first. This PR now only has some of the error handling and the resourcebuilder + improved linting

**Comment by JoelSpeed** [comment]
/retest-required

Something went wrong with the setup there, looks like the build of the container image failed

File: cmd/machine-api-operator/start.go [hunk 3]
```diff
@@ -139,7 +152,12 @@ func startControllers(ctx *ControllerContext) {
 		ctx.ClientBuilder.MachineClientOrDie(componentName),
 		ctx.ClientBuilder.DynamicClientOrDie(componentName),
 		recorder,
-	).Run(1, ctx.Stop)
+	)
+	if err != nil {
+		panic(fmt.Errorf("error creating operator: %v", err))
+	}
+
+	optr.Run(1, ctx.Stop)
```
Comment (JoelSpeed) [inline cmd/machine-api-operator/start.go:160]: This used to be handled in a go routine right? I think this is blocking now and causing the operator to fail

**Comment by RadekManak** [comment]
Looks like the added error checking exposed an error in machineset controller
`2023-02-09T12:48:59.169851734Z F0209 12:48:59.169705       1 main.go:58] failed to set logtostderr flag: no such flag -logtostderr`

**Pull request merged**